src/Sulu/Bundle/PageBundle/Sitemap/PagesSitemapProvider.php

```diff
@@ -85,6 +94,17 @@ public function build($page, $scheme, $host)
                     continue;
                 }
 
+                $userPermissions = $this->accessControlManager->getUserPermissionByArray(
+                    $contentPage->getLocale(),
+                    PageAdmin::SECURITY_CONTEXT_PREFIX . $contentPage->getWebspaceKey(),
+                    $contentPage->getPermissions(),
```

Comment:
It would be quite hard to write a correct query, because in PHPCR we have only the `sec:role-<id>` properties, and we don't know which of the roles are for the correct system.

But this is also not such a big problem, because the `$contentPage` already contains the permissions, as it has already loaded the entire document anyway. And the `getUserPermissionByArray` method only uses that information, and does not do any requests to the `se_access_controls` table (the access control for pages are also not stored there).

Reply:
Forgot about that Document has the Permissions directly. So this is no problem for me than.